### PR TITLE
major performance update to creating question table

### PR DESCRIPTION
**Pull request by bnmnetp**
This is really just about a small refactoring of how the questions table is constructed.

A somewhat sad side effect is that my refactoring tool decided to reformat the js file according to some guidelines that makes it look like I changed a lot more than I did.   Its nice to be consistent but these kinds of changes cause ripple annoying merge conflicts sometimes.

### CODE REVIEW

**Comment by bjones1** [comment]
Next time around, it would be nice to see this as two commits -- one with the reformat, the other with changes.

That being said, I really like the reformatting feature in your editor. If you ever have time, I'd suggest applying this to all the JS (and perhaps HTML) to clean it up.

File: static/js/admin.js [hunk 1]
```diff
@@ -1307,14 +1313,28 @@ function assignmentInfo() {
...
+        //     _id: _id,
+        // }
+        let allQuestions = []
         for (let question of data['questions_data']) {
             // Put the qeustion in the table.
             let name = question['name'];
-            appendToQuestionTable(name, question['points'], question['autograde'], question['autograde_possible_values'], question['which_to_grade'], question['which_to_grade_possible_values']);
+            allQuestions.push(createQuestionObject(name, question['points'], question['autograde'], question['autograde_possible_values'], question['which_to_grade'], question['which_to_grade_possible_values']));
             // Check this question in the question tree picker.
             tqp.check_node(tqp.get_node(name));
```
[the remark this answers is not in the document]
Comment (bnmnetp) [reply static/js/admin.js]: @bjones1 are you suggesting a for loop to add everything to ``allQuestions`` then call load and then another loop to call tqp.check_node for each name?

[the remark this answers is not in the document]
Comment (bjones1) [reply static/js/admin.js]: Yes, you're correct -- I was assuming this wouldn't work before adding the items, but if the pre-populated tree makes it work, great! So, you can safely ignore this comment.

You might add a comment to the code explaining why this works, though.

[the remark this answers is not in the document]
Comment (bnmnetp) [reply static/js/admin.js]: I'll do that.  Thanks for the review, and getting me pointed in the right direction.

[the remark this answers is not in the document]
Comment (bjones1) [reply static/js/admin.js]: Thanks for the fix!

**Pull request merged**